### PR TITLE
Replace react-autosuggest with downshift

### PR DESCRIPTION
**Pull request by jbalboni**
This seems to work much better and you can actually choose options in VoiceOver now.

screen shot 2018-01-04 at 4 50 42 pm

### CODE REVIEW

**Review by cehsu — APPROVED** [review]

**Review by rlbaker — APPROVED** [review]

File: src/js/common/schemaform/fields/AutosuggestField.jsx [hunk 1]
```diff
@@ -156,24 +148,46 @@ export default class AutosuggestField extends React.Component {
...
+          getInputProps,
+          getItemProps,
+          isOpen,
+          selectedItem,
+          highlightedIndex
+        }) => (
+          <div className="autosuggest-container">
+            <input {...getInputProps({ id, name: id, onKeyDown: this.handleKeyDown, onBlur: this.handleBlur })}/>
+            {isOpen && (
+              <div className="autosuggest-list">
```
Comment (cehsu) [inline src/js/common/schemaform/fields/AutosuggestField.jsx]: Running aXe coconut indicates that this div should be given the listbox role:
>Certain ARIA roles must contain particular children
Issue description
Ensures elements with an ARIA role that require child roles contain them

>Impact: critical
 Learn more
Element location
#root_application_veteran_serviceRecords_0_serviceBranch
Element source
<input role="combobox" aria-autocomplete="list" aria-expanded="false" autocomplete="off" value="army" id="root_application_veteran_serviceRecords_0_serviceBranch" name="root_application_veteran_serviceRecords_0_serviceBranch">

>To solve this violation, you need to:
Fix the following:
Required ARIA child role not present: listbox

File: src/sass/modules/_m-schemaform.scss [hunk 2]
```diff
@@ -508,11 +508,11 @@ legend.schemaform-label.schemaform-file-label {
   margin-top: 0;
 }
 
-.react-autosuggest__container {
+.autosuggest-container {
```
Comment (cehsu) [inline src/sass/modules/_m-schemaform.scss]: It looks like there might be a styling issue on mobile:
screen shot 2018-01-04 at 4 12 36 pm

**Comment by jbalboni** [comment]
@cehsu Both those issues should be fixed now.

**Review by bshyong — APPROVED** [review]
nice!

**Pull request merged**